inspirehep/modules/editor/views.py

```diff
@@ -225,3 +228,38 @@ def _simplify_ticket_response(ticket):
         owner=ticket['Owner'],
         date=ticket['Created'],
         link=ticket['Link'])
+
+
+@blueprint.route('/<recid>/upload', methods=['POST'])
+@editor_permission
+def upload_files(recid):
+    # check if the post request has the file part
+    if 'file' not in request.files:
+        return jsonify(success=False, message='No file part')
+    file = request.files['file']
+    # if user does not select file, browser also
+    # submit a empty part without filename
+    if file.filename == '':
+        return jsonify(success=False, message='No selected file')
+    if file:
+        filedir = os.path.join(current_app.config['RECORD_EDITOR_FILE_UPLOAD_FOLDER'], recid)
+        if not os.path.isdir(filedir):
+            os.makedirs(filedir)
+        filename = secure_filename(file.filename)
+        filepath = os.path.join(
+            filedir,
+            filename
+        )
+        file.save(filepath)
+        return jsonify(filepath)
+
+
+@blueprint.route('/<recid>/upload/clean', methods=['DELETE'])
+@editor_permission
+def clean_files(recid):
```

Comment:
better to do cron job, otherwise another weird logic on the frontend.
@jacquerie @david-caro @michamos